### PR TITLE
2117 poll for new heads doc updates

### PR DESCRIPTION
**Pull request by ebadiere**
Updated documentation.  Includes design and configuration updates.

**Related issue(s)**: #2117

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ Unless you need to set a non-default value, it is recommended to only populate o
 | `WS_CONNECTION_LIMIT_PER_IP`    | "10"     | Maximum amount of connections from a single IP address                                                                        |
 | `WS_MULTIPLE_ADDRESSES_ENABLED` | "false"  | If enabled eth_subscribe will allow subscription to multiple contract address.                                                |
 | `WS_CACHE_TTL`                  | "20000"  | The time to live for cached entries.                                                                                          |
+| `WS_NEW_HEADS_ENABLED`.         | "false".  | Enables subscriptions for the lates blocks, `newHeads`.                                                                      |
```
Comment (Nana-EC) [inline docs/configuration.md:116]:
```suggestion
| `WS_NEW_HEADS_ENABLED`.         | "false".  | Enables subscriptions for the latest blocks, `newHeads`.                                                                      |
```

File: docs/design/eth-subscribe.md [hunk 2]
```diff
@@ -200,6 +206,27 @@ if (success) {
 
 3. Create a scheduled job that loops through all saved subscriptions and calls the Mirror node Rest API and sends the response to all `subscribers`. It should be possible to configure the time interval. A throttling mechanism can be implemetented that limits the consecutive calls to the Mirror node, and at every interval only X subscriptions with the oldest `lastUpdated` should be polled.
 4. Whenever a connection is closed all corresponding subscriptions should be deleted from memory.
+   
+### Poll for Blocks, `newHeads` Implementation 
+The `newHeads` implementation simply builds on the existing mechanism for polling for logs.  The differences with the polling for logs is as follows:
+1. Add a condition check for `newHeads`, `validateSubscribeEthNewHeads`, and a `newHeads` subscription to the `webSocketServer.ts`.
+2. Add a call to the relay subscribe with the socket, event, and filter. 
+3. Add a call to the mirror node in the `poller.ts` to handle the `newHeads` event by calling `eth_getBlockByNumber` for `latest`.
```
Comment (Nana-EC) [inline docs/design/eth-subscribe.md:214]: Q: is this a mistake? Did you mean "relay"?

File: docs/design/eth-subscribe.md [hunk 3]
```diff
@@ -200,6 +206,27 @@ if (success) {
...
+4. Add logging where appropriate.  We can add metrics in the `poller.ts` for the call to the relay as it will call the mirror node to get the latest block.
```
Comment (Nana-EC) [inline docs/design/eth-subscribe.md:215]: Q: What metrics should be added?

**Review by Nana-EC — COMMENTED** [review]
Looking good a few nits

File: docs/design/eth-subscribe.md (hunk 2 above)
Comment (ebadiere) [reply docs/design/eth-subscribe.md:214]: Yes, I'll remove "the mirror node", it's confusing.  I had the relay querying the mirror node when running `eth_getBlocksByNumber` when I wrote that.

File: docs/design/eth-subscribe.md (hunk 3 above)
Comment (ebadiere) [reply docs/design/eth-subscribe.md:215]: The time it takes for each `eth_getBlocksByNumber` to complete, to monitor latency.  I'll make the update.

File: docs/design/eth-subscribe.md (hunk 2 above)
Comment (ebadiere) [reply docs/design/eth-subscribe.md:214]: Done.

File: docs/design/eth-subscribe.md (hunk 3 above)
Comment (ebadiere) [reply docs/design/eth-subscribe.md]: Done.

File: docs/design/eth-subscribe.md [hunk 4]
```diff
@@ -9,12 +9,12 @@ The JSON-RPC Relay currently doesn't support subscription using websocket as is
 1. Implement websocket subscription in the relay
 2. Introduce two new methods - `eth_subscribe` and `eth_unsubscribe`
 3. Support subscription for event logs
+4. Support subscription for latest blocks `newHeads`
 
 ## Non-Goals
 
 1. Add long-term memory to the relay
 2. Reduce the efficiency of the relay or mirror node due to high resource usage
```
Comment (acuarica) [inline docs/design/eth-subscribe.md:17]: Maybe not related but "reduce the efficiency"?

File: docs/design/eth-subscribe.md [hunk 5]
```diff
@@ -38,12 +38,18 @@ wscat -c wss://<env>.hashio.io/api
...
+{"id":1,"jsonrpc":"2.0","method":"eth_subscribe","params":["newHeads"]}
 ```
 
 #### Response
 
 ```javascript
 {"jsonrpc":"2.0","id": 1, "result": SUBSCRIPTION_ID}
+
+// newHeads
+{"jsonrpc":"2.0","id": 1, "result": SUBSCRIPTION_ID}
```
Comment (acuarica) [inline docs/design/eth-subscribe.md:52]: Should we show some example of `SUBSCRIPTION_ID`, like Geth does https://geth.ethereum.org/docs/interacting-with-geth/rpc/pubsub?

**Review by acuarica — APPROVED** [review]
lgtm some nits

File: docs/design/eth-subscribe.md (hunk 4 above)
Comment (ebadiere) [reply docs/design/eth-subscribe.md:17]: This was written as an Non-Goal for the original `eth_subscribe` for logs implementation.  I added the requirement to monitor calls to `eth_getBlockByNumber` to detect latency in the `newHeads` feature.

**Pull request merged**